boards/xtensa/intel_adsp_cavs15/doc/index.rst

```diff
@@ -72,55 +72,80 @@ Loading image to Audio DSP
 `SOF Diagnostic Driver`_ provide interface for firmware loading. Python tools
 in the board support directory use the interface to load firmware to ``ADSP``.
 
-Note that the ``/dev/hda`` device file created by the diagnostic
-driver must be readable and writable by the process.  This can be
-accomplished via a simple chmod, via a udev handler that associates
-the device with a particular user or group, or simply by running the
+Assume that the up_squared board's host name is ``cavs15`` (It also can be an
+ip address), and the user account is ``user``. Then copy the python tool to the
+``up_squared`` board from your building evironment:
+
+   $scp boards/xtensa/intel_adsp_cavs15/tools/cavstool_server.py user@cavs15:
+
+
+Note that the ``/dev/hda`` device file created by the diagnostic driver must
+be readable and writable by the process.  So we simply by running the
 loader script as root:
 
 .. code-block:: console
 
-   $ sudo chmod 777 /dev/hda
-   $ boards/xtensa/intel_adsp_cavs15/tools/fw_loader.py -f <path to zephyr.ri>
+   cavs15$ sudo ./cavstool_server.py
 
-Debugging
-=========
+It's a daemon that is in change of loading the firmware to ADSP, and send out the
+log message.
 
-The only way to debug application is using logging. Logging and ADSP logging
-backend needs to be enabled in the application configuration.
+Running and Debugging
+=====================
 
-ADSP logging backend writes logs to the ring buffer in the shared memory.
+While the python script is running on ``up_squared`` board, you can start load
+image and run the application by:
 
-As above, the ``adsplog`` tool requires appropriate permissions, in
-this case to the sysfs "resource4" device on the appropriate PCI
-device.  This can likewise be managed via any filesystem, setuid or
-udev trick the operator prefers.
+.. code-block:: console
+
+   west flash --remote-host cavs15
```

Comment:
No, if we have: 
(1) rimage tool install in system path (such as: /usr/bin or /usr/local/bin ..) 
(2) config file:  If we follow the SOF's setup steps this will be in modules/audio/sof/rimage/config
(3) key:  the default key location will be: modules/audio/sof/keys.
Then west flash --remote-host will be only parameter needed.